[templates] Move project templates from the universe repo

tsapeta opened the pull request:
Moving `expo-template-blank` and `expo-template-tabs` from the universe repo, so all project templates will be at the same place.
Unfortunately there are some duplicated commits because I wanted to keep the history and was unable to move these two folders at once.

Also added `publish-project-templates` script to `expotools`. Demo: 
![Screen Shot 2019-05-30 at 09 54 05](https://user-images.githubusercontent.com/1714764/58618307-8e0d1880-82c2-11e9-871a-efef88d4222a.png)

fson committed:
[templates] Rename project templates and publish them to npm (#3313)

fson committed:
[templates] Rename project templates and publish them to npm (#3313)

fson committed:
[templates] Add empty lines

fson committed:
[templates] Add empty lines

quinlanj committed:
[expo-template-tabs] apply changes from watanabeyu

tsapeta committed:
[templates] update blank template for SDK32

sjchmiela committed:
[templates] Update templates for SDK32 (#3356)

* [templates] Update expo-template-tabs to SDK32 (tests failing)

* [expo-template-tabs] Upgrade dependencies

* [expo-template-tabs] Upgrade jest-expo

* [expo-template-tabs] Upgrade jest-expo

* [expo-template-blank] Update dependencies to published SDK32

* [expo-template-tabs] Update dependencies to published SDK32

* [templates] Publish templates for SDK32

sjchmiela committed:
[templates] Update templates for SDK32 (#3356)

* [templates] Update expo-template-tabs to SDK32 (tests failing)

* [expo-template-tabs] Upgrade dependencies

* [expo-template-tabs] Upgrade jest-expo

* [expo-template-tabs] Upgrade jest-expo

* [expo-template-blank] Update dependencies to published SDK32

* [expo-template-tabs] Update dependencies to published SDK32

* [templates] Publish templates for SDK32

dsokal committed:
[expo-init-templates] add p8 files to .gitignore

dsokal committed:
[expo-init-templates] add p8 files to .gitignore

fson committed:
Remove alpha channel from icons (#3456)

fson committed:
Remove alpha channel from icons (#3456)

EvanBacon committed:
Added web support to templates (#3486)

EvanBacon committed:
Added web support to templates (#3486)

EvanBacon committed:
Updated git ignore in templates

EvanBacon committed:
Updated git ignore in templates

sergeichestakov committed:
Add .orig to gitignore template (#3522)

sergeichestakov committed:
Add .orig to gitignore template (#3522)

ide committed:
[template] Use a function component in blank template's App.js

The blank template is intended to be minimal -- removing the class extension slims this down further.

ide committed:
[template] Optimize PNG images for tabs template

Optimized the PNG images further for performance and to use less bandwidth.

ide committed:
[template] Update blank template for SDK 33

ide committed:
[template] Update tabs template to SDK 33

ide committed:
[template] Update tabs template for SDK 33, use hooks

This updates the tabs template to work with SDK 33 (dependency versions are not yet final) and uses modular imports. It uses React hooks as well. I also made the unit tests render a shallow tree so that the unit under test is smaller.

Test Plan:

- Run Jest tests with `yarn jest`
- Open project on iOS after setting sdkVersion to 33.0.0 and verify that all screens and links work

ide committed:
[template] Use Reat 16.8.3 instead of 16.8.6 in the blank template

Use the precise version, since sometime patch versions of React have broken the interface between React and RN.

tsapeta committed:
[templates] move expo-template-blank to templates folder

tsapeta committed:
Merge branch '@tsapeta/expo-template-blank' into @tsapeta/move-templates-from-universe

tsapeta committed:
[templates] move expo-template-tabs to templates folder

tsapeta committed:
Merge branch '@tsapeta/expo-template-tabs' into @tsapeta/move-templates-from-universe

ide commented:
This TS template also needs to be moved over sometime: https://github.com/expo/universe/commit/32f60c7d145e05bae903d35d2c4844a897dd95ba

ccheever committed:
[templates] Adding the blank template but using TypeScript

Basically just copied the directory for the blank template
then followed the instructions for adding TypeScript to an Expo
project and then made a few tweaks based on discussions with @ide

tsapeta committed:
[expotools] add expotools command to publish project templates

sjchmiela commented on templates/expo-template-tabs/package.json:
```diff
@@ -0,0 +1,35 @@
+    "expo-asset": "^5.0.0-rc.0",
+    "expo-constants": "^5.0.0-rc.0",
+    "expo-font": "^5.0.0-rc.0",
+    "expo-web-browser": "^5.0.0-rc.0",
```
`^5.0.0`?

sjchmiela reviewed (commented):
One comment and please squash commits ❤️
